c10/util/Optional.h

```diff
@@ -42,20 +42,6 @@
 #include <c10/util/C++17.h>
 #include <c10/util/Metaprogramming.h>
 
-C10_CLANG_DIAGNOSTIC_PUSH()
-#if C10_CLANG_HAS_WARNING("-Wstring-conversion")
-C10_CLANG_DIAGNOSTIC_IGNORE("-Wstring-conversion")
-#endif
-#if C10_CLANG_HAS_WARNING("-Wshorten-64-to-32")
-C10_CLANG_DIAGNOSTIC_IGNORE("-Wshorten-64-to-32")
-#endif
-#if C10_CLANG_HAS_WARNING("-Wimplicit-float-conversion")
-C10_CLANG_DIAGNOSTIC_IGNORE("-Wimplicit-float-conversion")
-#endif
-#if C10_CLANG_HAS_WARNING("-Wimplicit-int-conversion")
-C10_CLANG_DIAGNOSTIC_IGNORE("-Wimplicit-int-conversion")
-#endif
```

Comment:
No action needed, but are you working on moving us to std::optional too?

Reply:
@swolchok has a work in progress. I am waiting that to finish.

Reply:
@ezyang But I have a work moving to std::variant #103675